improver/spotdata/spot_extraction.py

```diff
@@ -315,9 +320,17 @@ def process(
         spot_values = diagnostic_cube.data[..., y_indices, x_indices]
 
         additional_dims = None
+        additional_dims_aux = []
         if len(spot_values.shape) > 1:
             additional_dims = diagnostic_cube.dim_coords[:-2]
-
+            for dim_coord in additional_dims:
+                dim_coord_dim = diagnostic_cube.coord_dims(dim_coord)
+                aux_coords = [
```

Comment:
I think I would have done this the other way around, where the utility does this logic and assumes you want to copy forward all the aux coords associated with the additional_dims. I think this would be more robust to some cases that could occur but that we don't need to handle right now, like if you had more than one aux coord associated with a dim coord or more than one dim coord you are handling.

Reply:
Yes, I see your point. That would be a better place for this logic. I'll move it into `build_diagnostic_cube`.